src/idl_gen_fbs.cpp

```diff
@@ -136,8 +141,25 @@ std::string GenerateFBS(const Parser &parser, const std::string &file_name) {
         GenComment(field.doc_comment, &schema, nullptr, "  ");
         schema += "  " + field.name + ":" + GenType(field.value.type);
         if (field.value.constant != "0") schema += " = " + field.value.constant;
-        if (field.IsRequired()) schema += " (required)";
-        if (field.key) schema += " (key)";
+        std::vector<std::string> attributes;
+        if (field.IsRequired()) attributes.push_back("required");
+        if (field.key) attributes.push_back("key");
+
+        const auto &id_str = field.attributes.Lookup("id");
+        if (id_str && !id_str->constant.empty())
+          attributes.push_back("id: " + id_str->constant);
+		else
+		  attributes.push_back("id: " + std::to_string(counter++));
```

Comment:
This is not part of the final code. I have put it here to temporary fix map and union problem.